src/simulation/onboardDataHandling/storageUnit/partitionedStorageUnit.i

```diff
@@ -37,10 +37,10 @@ from Basilisk.architecture.swig_common_model import *
 
 //When using scientific notation in Python (1E9), it is interpreted as float
 // giving a type error when assigning storageCapacity or adding data through
-// setDataBuffer. This maps that float and vector of floats to long int in 
-// C++ in this module.
-%typemap(in) long long int {
-    $1 = static_cast<long long int>(PyFloat_AsDouble($input));
+// setDataBuffer. This maps that float and vector of floats to int64_t and 
+// long long int in C++ in this module.
+%typemap(in) int64_t {
+    $1 = static_cast<int64_t>(PyFloat_AsDouble($input));
 }
 %typemap(in) std::vector<long long int> (std::vector<long long int> temp) {
```

Comment:
Is this still needed following the change to `int64_t`?

Reply:
Unfortunately, yes. When setting the storage using scientific notation such as `1e3` in python, it will treat it as a float, leading to a `TypeError`. The second `typemap` in line 45 is in there for the same reason, but it uses `std::vector<long long int>` since it was causing some errors when switching to `std::vector<int64_t>`. I believe it is related to how swig works with the `typemap`, but I couldn't fix it.

Reply:
Let's create an issue for us to work out why this second typemap works/doesn't work when we would expect/hope. "It works" is rarely a good justification to go with code 😄

Reply:
Okay! I will create a new issue to address this specific typemap so we can try to figure out the problem.